src/test/java/com/artipie/PromuSliceITCase.java

```diff
@@ -0,0 +1,118 @@
+import org.junit.jupiter.api.io.TempDir;
+
+/**
+ * IT tests for Prometheus metrics.
+ * @since 0.23
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+final class PromuSliceITCase {
+
+    @Test
+    void givesAccessWhenPrometheusConfigIsAvailable(
+        @TempDir final Path root) throws Exception {
+        final int port = runServer(
+            root,
+            Yaml.createYamlMappingBuilder()
+                .add(
+                    "metrics",
+                    Yaml.createYamlMappingBuilder()
+                        .add("type", "prometheus")
+                        .build()
+                )
+        );
+        final HttpURLConnection con = (HttpURLConnection) new URL(formatUrl(port))
+            .openConnection();
+        con.setRequestMethod(HttpMethod.GET);
+        con.setRequestProperty(Accept.NAME, MediaType.TEXT_PLAIN);
+        MatcherAssert.assertThat(
+            con.getResponseCode(),
+            new IsEqual<>(
+                Integer.parseInt(RsStatus.OK.code())
+            )
+        );
+        con.disconnect();
+    }
+
+    @Test
+    void blocksAccessWhenPrometheusConfigIsUnavailable(
+        @TempDir final Path root) throws Exception {
+        final int port = runServer(
+            root,
+            Yaml.createYamlMappingBuilder()
+        );
+        final HttpURLConnection con = (HttpURLConnection) new URL(formatUrl(port))
+            .openConnection();
+        con.setRequestMethod(HttpMethod.GET);
+        con.setRequestProperty(Accept.NAME, MediaType.TEXT_PLAIN);
+        MatcherAssert.assertThat(
+            con.getResponseCode(),
+            new IsEqual<>(
+                Integer.parseInt(RsStatus.NOT_FOUND.code())
+            )
+        );
+        con.disconnect();
+    }
+
+    /**
+     * Runs an Artipie server.
+     * @param root Server root folder
+     * @param meta Meta config
+     * @return Port on which the server starts
+     */
+    private static int runServer(
+        final Path root, final YamlMappingBuilder meta) throws Exception {
+        final Vertx vertx = Vertx.vertx();
+        final Path repos = root.resolve("repos");
+        repos.toFile().mkdir();
+        final Path cfg = root.resolve("artipie.yaml");
+        Files.write(
+            cfg,
+            Yaml.createYamlMappingBuilder()
+                .add(
+                    "meta",
+                    meta.add(
+                        "storage",
+                        Yaml.createYamlMappingBuilder()
+                            .add("type", "fs")
+                            .add("path", repos.toString())
+                            .build()
+                    ).build()
+                )
+                .build().toString().getBytes()
+        );
+        final JettyClientSlices http = new JettyClientSlices(new HttpClientSettings());
+        http.start();
+        final VertxMain server = new VertxMain(http, cfg, vertx, 0);
+        return server.start();
```

Comment:
Each resource should be started in `setup` and stopped in `tearDown` test method - this way we can control the lifecycle. Right now, this test produces a few different servers without stopping it.

Reply:
Right !

Reply:
Besides, how about using `testcontainers` as @olenagerasimova mentioned early ?